workers/tests/integrationtests/api_response_schema.py

```diff
@@ -49,19 +32,75 @@ def get_response_schema() -> Schema:
     )
 
 
-def get_feature_schema(iterator: Optional[str] = None):
+def get_result_schema() -> Schema:
+    """Response schema for all endpoints except of 'indicator' and 'report' endpoints"""
+    return Schema({"result": list}, ignore_extra_keys=True)
+
+
+def get_featurecollection_schema() -> Schema:
+    """Response schema for responses of type FeatureCollection"""
+    return Schema(
+        {"type": "FeatureCollection", "features": list}, ignore_extra_keys=True
+    )
+
+
+def get_indicator_feature_schema() -> Schema:
     return Schema(
         {
             "type": "Feature",
             "geometry": dict,
             Opt("id"): Or(str, int),
-            "properties": generate_properties_template(iterator),
+            "properties": {
+                "metadata.name": str,
+                "metadata.description": str,
+                "layer.name": str,
+                "layer.description": str,
+                "result.timestamp_oqt": str,
+                "result.timestamp_osm": Or(str),
+                "result.value": Or(float, None),
+                "result.label": str,
+                "result.description": str,
+                Opt("result.svg"): str,
+            },
         },
         ignore_extra_keys=True,
     )
 
 
-def get_featurecollection_schema():
-    return Schema(
-        {"type": "FeatureCollection", "features": list}, ignore_extra_keys=True
+def get_report_feature_schema(number_of_indicators: int) -> Schema:
+    properties_template = {
+        "metadata.name": str,
+        "metadata.description": str,
+        "layer.name": str,
+        "layer.description": str,
+        "result.timestamp_oqt": str,
+        "result.timestamp_osm": Or(str),
+        "result.value": Or(float, None),
+        "result.label": str,
+        "result.description": str,
+        "result.svg": str,
+    }
+    properties = {}
+    for i in range(number_of_indicators):
+        for k, v in properties_template.items():
+            if k == "result.svg":
+                properties.update({Opt("indicators." + str(i) + "." + k): v})
+            else:
+                properties.update({"indicators." + str(i) + "." + k: v})
```

Comment:
Why do you double the `properties_template` code here, if you overwrite `"result.svg"` anyway?

Reply:
This code smell is related to this issue: https://sonarcloud.io/project/issues?branch=PR-168&id=ohsome-quality-analyst&open=AXu650vLVi-dpvTMVG1C&resolved=false&types=CODE_SMELL